internal/pkg/admission/policy-server-configmap.go

```diff
@@ -40,35 +41,60 @@ func (r *Reconciler) reconcilePolicyServerConfigMap(
 		return fmt.Errorf("cannot lookup policies ConfigMap: %w", err)
 	}
 
-	newPoliciesYML, err := r.createPoliciesYML(clusterAdmissionPolicies)
+	return r.updateIfNeeded(ctx, cfg, clusterAdmissionPolicies)
+}
+
+func (r *Reconciler) updateIfNeeded(ctx context.Context, cfg *corev1.ConfigMap, clusterAdmissionPolicies *policiesv1alpha2.ClusterAdmissionPolicyList) error {
+	newPoliciesMap, err := r.createPoliciesMap(clusterAdmissionPolicies)
 	if err != nil {
 		return fmt.Errorf("cannot create policies: %w", err)
 	}
-	currentPoliciesYML := cfg.Data[constants.PolicyServerConfigPoliciesEntry]
 
-	if currentPoliciesYML != newPoliciesYML {
+	if shouldUpdate, err := shouldUpdatePolicyMap(cfg.Data[constants.PolicyServerConfigPoliciesEntry], newPoliciesMap); err != nil {
+		return fmt.Errorf("cannot compare policies: %w", err)
+	} else if shouldUpdate {
+		newPoliciesYML, err := json.Marshal(newPoliciesMap)
+		if err != nil {
+			return fmt.Errorf("cannot marshal policies: %w", err)
+		}
 		patch := cfg.DeepCopy()
-		patch.Data[constants.PolicyServerConfigPoliciesEntry] = newPoliciesYML
+		patch.Data[constants.PolicyServerConfigPoliciesEntry] = string(newPoliciesYML)
 		err = r.Client.Patch(ctx, patch, client.MergeFrom(cfg))
 		if err != nil {
 			return fmt.Errorf("cannot patching policies: %w", err)
 		}
 	}
+
 	return nil
 }
 
+func shouldUpdatePolicyMap(currentPoliciesYML string, newPoliciesMap map[string]policyServerConfigEntry) (bool, error) {
+	var currentPoliciesMap map[string]policyServerConfigEntry
+
+	if err := json.Unmarshal([]byte(currentPoliciesYML), &currentPoliciesMap); err != nil {
+		return false, fmt.Errorf("cannot unmarshal policies: %w", err)
+	}
+
+	return !reflect.DeepEqual(currentPoliciesMap, newPoliciesMap), nil
```

Comment:
I like this solution a lot, thanks @raulcabello. I cannot foresee any problems at this time.